src/tools/perf/lib/ucp_tests.cc

```diff
@@ -1031,12 +968,15 @@ class ucp_perf_test_runner {
     ucp_atomic_op_t     m_atomic_op;
 };
 
-
 #define TEST_CASE(_perf, _cmd, _type, _flags, _mask) \
     if (((_perf)->params.command == (_cmd)) && \
         ((_perf)->params.test_type == (_type)) && \
         (((_perf)->params.flags & (_mask)) == (_flags))) \
     { \
+        if ((_perf)->params.send_device.mem_type != UCS_MEMORY_TYPE_LAST) { \
+            ucp_perf_test_device_runner<_cmd, _type, _flags> r(*_perf); \
+            return r.run(); \
+        } \
```

Comment:
i don't think we can instantiate the runner here, since it will AFAIU include direct calls to device API, so it should be part of the "cuda" module